sktalk/corpus/conversation.py

```diff
@@ -8,8 +13,22 @@ def __init__(
             utterances (list[Utterance]): A list of Utterance objects representing the utterances in the conversation.
             metadata (dict, optional): Additional metadata associated with the conversation. Defaults to None.
         """
+        self._metadata = metadata or {}
         self._utterances = utterances
-        self._metadata = metadata
+        # Input utterances should be a list of type Utterance
+        errormsg = "All utterances in a conversation should be of type Utterance"
+        if not isinstance(self._utterances, list):
+            try:
+                self._utterances = list(self._utterances)
+            except TypeError as e:
+                raise TypeError(errormsg) from e
```

Comment:
Perhaps make possible and allowed cases more explicit.

Reply:
Thanks for the comment; I like your approach, but I am opting to keep it this way. I set this up to be intentionally gregarious: if a conversation is entered as a single Conversation object, it will be put in a list. If, for some reason, the Conversations are bundled in a tuple or a dictionary, they are also turned into a list. I do not think these are realistic scenarios, but should they occur (and later pass the verification of the content of the list), it is nice to have them processed instead of fail at import.
If there is an integer here, for instance, the `list()` operation will fail already. Hence the try/except clause.

Before continuing, the content of the list is verified. Every item must be a Conversation. If not, the import will fail.